### DISCUSSION TITLE
Does anyone upgrade Flux using Renovate?

### DISCUSSION THREAD

**Discussion by loganmarchione** [fluxcd/flux2#3354] [Q&A] [answered]
This might be a dumb question, but I didn't see it quite spelled out in the [docs](https://fluxcd.io/flux/installation/#upgrade).

I'm using Flux to manage a K3s cluster, and I use [Renovate](https://github.com/renovatebot/renovate) to manage my dependencies. If you don't know, Renovate is a dependency management tool (like GitHub's Dependabot).

1. Renovate constantly scans my repo for out-of-date dependencies
2. If it finds anything, Renovate opens a PR to upgrade the version of my containers
3. Flux scans the repo, sees the update, and redeploys my applications

I have a PR (opened by Renovate) to upgrade Flux from `v0.36.0` to `v0.37.0`. Here is the [link to the PR itself](https://github.com/loganmarchione/k8s_homelab/pull/29/files).

Is it safe to merge this PR, or am I better off upgrading Flux using the bootstrap command?

**Comment by AndersBennedsgaard** [comment, marked as answer]
I also just added Renovate to a Flux-based repository, and got the same PR: v0.36.0 -> v0.37.0.
I tested the upgrade by checkout out the main branch, run
```bash
flux install --components-extra="image-reflector-controller,image-automation-controller" --export > gotk-components.yaml
```
and checking the diff on `gotk-components.yaml` between my changes and the changes by Renovate. It returned nothing, which means that Renovate probably use the exact same command for upgrading Flux.
That is probably also why the Flux CLI must be present for Renovate to be able to upgrade the components.

**Reply by loganmarchione** [reply]
Thanks for testing this!
